.github/workflows/third-party-library-changes.yml

```diff
@@ -0,0 +1,63 @@
+name: Third Party Library Changes
+
+on:
+  pull_request:
+    paths:
+      - 'src/wp-includes/ID3/**'
+      - 'src/wp-includes/IXR/**'
+      - 'src/wp-includes/PHPMailer/**'
+      - 'src/wp-includes/Requests/**'
+      - 'src/wp-includes/SimplePie/**'
+      - 'src/wp-includes/sodium_compat/**'
+      - 'src/wp-includes/Text/**'
+
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
+jobs:
+  third-party-changed-files:
+    runs-on: ubuntu-latest
+    name: Get third party library modified files
+    permissions:
+      issues: write
+      pull-requests: write
+    if: ${{ github.repository == 'johnbillion/wordpress-develop' }}
+    steps:
+      - name: Leave a comment about third party libraries
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            const { owner, repo } = context.repo;
+            const { number } = context.issue;
+
+            // Check for the presence of a comment and bail early.
+            const comments = ( await github.rest.issues.listComments( { owner, repo, issue_number: number } ) ).data;
+
+            const hasComment = comments.some( comment =>
+              comment.user.type === 'Bot' && comment.body.includes( 'Third Party Files' )
+            );
+
+            if ( hasComment ) return;
+
+            github.rest.issues.createComment( {
+              owner,
+              repo,
+              issue_number: number,
+              body: `## Third Party Files
+            This pull request modifies a third party library file. Changes to files in the following directories should be made to their respective upstream projects instead:
```

Comment:
A good follow up item, but I wonder if we could extract these messages into `.md` files and pull them in instead of including the comment contents inline. Could make editing and tracking updates much easier. But may not help if there's a need for `${{ github.context }}` style substitutions.